src/common/managers/SceneManager.hpp

```diff
@@ -0,0 +1,150 @@
+/*
+** EPITECH PROJECT, 2024
+** StellarForge
+** File description:
+** SceneManager.hpp
+*/
+
+#pragma once
+
+#include <map>
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include "../IScene.hpp"
+#include "../UUID.hpp"
+#include "ManagerException.hpp"
+
+/**
+ * @class SceneManager
+ * @brief Manages scenes in the StellarForge engine.
+ * @details This class provides an interface for managing multiple scenes in the application. It allows adding, removing, and switching between scenes.
+ * @version v0.1.0
+ * @since v0.1.0
+ * @author Aubane NOURRY
+ */
+class SceneManager {
+public:
+    /**
+     * @brief Default constructor for SceneManager.
+     * @details Initializes the SceneManager with no scenes and sets the current scene index to -1.
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    SceneManager() : _currentSceneIndex(-1) {}
+
+    /**
+     * @brief Destructor for SceneManager.
+     * @details Cleans up the SceneManager when it is no longer in use.
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    ~SceneManager() = default;
+
+    /**
+     * @brief Adds a scene to the manager.
+     * @param uuid The unique identifier of the scene.
+     * @param scene A shared pointer to the scene to be added.
+     * @param position The position to add the scene at. Default is -1, which appends the scene.
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Aubane NOURRY
+     */
+    void addScene(const UUID& uuid, const std::shared_ptr<IScene> &scene, int position /*default -1*/);
```

Comment:
Oh and just like ObjectManager I believe that the uuid could be generated by the scene manager